eradicate.py

```diff
@@ -28,176 +28,199 @@
+        for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
+                                           'import']:
+            if symbol in line:
+                break
+        else:
+            return False
 
-MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
-PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
-
-
-def comment_contains_code(line, aggressive=True):
-    """Return True comment contains code."""
-    line = line.lstrip()
-    if not line.startswith('#'):
-        return False
-
-    line = line.lstrip(' \t\v\n#').strip()
-
-    # Ignore non-comment related hashes. For example, "# Issue #999".
-    if re.search('#[0-9]', line):
-        return False
-
-    if line.startswith('pylint:'):
-        return False
+        if self.multiline_case(line, aggressive=aggressive):
+            return True
 
-    if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
-        return False
+        symbol_list = [r'elif\s+.*', 'else', 'try',
+                       'finally', r'except\s+.*']
+        if aggressive:
+            symbol_list.append(r'if\s+.*')
 
-    # Check that this is possibly code.
-    for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
-                                       'import']:
-        if symbol in line:
-            break
-    else:
-        return False
+        for symbol in symbol_list:
+            if re.match(r'^\s*' + symbol + r'\s*:\s*$', line):
+                return True
 
-    if multiline_case(line, aggressive=aggressive):
-        return True
+        line = re.sub(r'^(print|return)\b\s*', '', line)
 
-    symbol_list = [r'elif\s+.*', 'else', 'try',
-                   'finally', r'except\s+.*']
-    if aggressive:
-        symbol_list.append(r'if\s+.*')
+        if re.match(self.PARTIAL_DICTIONARY_REGEX, line):
+            return True
 
-    for symbol in symbol_list:
-        if re.match(r'^\s*' + symbol + r'\s*:\s*$', line):
+        try:
+            compile(line, '<string>', 'exec')
             return True
+        except (SyntaxError, TypeError, UnicodeDecodeError):
+            return False
 
-    line = re.sub(r'^(print|return)\b\s*', '', line)
 
-    if re.match(PARTIAL_DICTIONARY_REGEX, line):
-        return True
+    def multiline_case(self, line, aggressive=True):
+        """Return True if line is probably part of some multiline code."""
+        if aggressive:
+            for ending in ')]}':
+                if line.endswith(ending + ':'):
+                    return True
 
-    try:
-        compile(line, '<string>', 'exec')
-        return True
-    except (SyntaxError, TypeError, UnicodeDecodeError):
-        return False
+                if line.strip() == ending + ',':
+                    return True
 
+            # Check whether a function/method definition with return value
+            # annotation
+            if re.search(r"def .+\)[\s]+->[\s]+[a-zA-Z_][a-zA-Z0-9_]*:$", line):
```

Comment:
These are all constants.
We can also use `re.compile` early.

Reply:
Put them into `DEF_STATEMENT_REGEX`, `WITH_STATEMENT_REGEX` and  `FOR_STATEMENT_REGEX`